[AMBARI-22886] Infra Manager: store s3 credentials in Hadoop credential store

kasakrisz opened the pull request:
## What changes were proposed in this pull request?

- store s3 access key id and secret key to hadoop credential store
- validate job properties only if job enabled
- do not schedule job if validation failed

## How was this patch tested?

- UTs passed
- manual testing using ambari

Please review
@oleewere @swagle @zeroflag

kasakrisz committed:
AMBARI-22886 Infra Manager: store s3 credentials in Hadoop credential store

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-Github-PullRequest-Builder/428/
Test PASSed.

adoroszlai commented on ambari-infra/ambari-infra-manager/src/main/java/org/apache/ambari/infra/conf/security/CompositePasswordStore.java:
```diff
@@ -0,0 +1,42 @@
+  private List<PasswordStore> passwordStores;
+
+  public CompositePasswordStore(PasswordStore... passwordStores) {
+    this.passwordStores = new ArrayList<>(asList(passwordStores));
```
Why create a new `ArrayList` if its size never changes?

adoroszlai commented on ambari-infra/ambari-infra-manager/src/test/java/org/apache/ambari/infra/conf/security/CompositePasswordStoreTest.java:
```diff
@@ -0,0 +1,47 @@
+  }
+
+  @Test
+  public void testGetPasswordReturnPasswordFromFistStoreIfExists() {
```
typo: `FistStore` -> `FirstStore`

adoroszlai commented on ambari-infra/ambari-infra-manager/src/main/java/org/apache/ambari/infra/conf/security/PasswordStore.java:
```diff
@@ -0,0 +1,23 @@
+package org.apache.ambari.infra.conf.security;
+
+public interface PasswordStore {
+  String getPassword(String propertyName);
```
Since this is a new interface, it may be safer to define return type as `Optional<String>` instead.

kasakrisz committed:
AMBARI-22886 Infra Manager: store s3 credentials in Hadoop credential store
- addendum: use Optional<String> as a return type of getPassword, remove unnecessary ArrayList

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-Github-PullRequest-Builder/444/
Test PASSed.
